docs/nnx/nnx_basics.md

```diff
@@ -376,3 +376,52 @@ model = nnx.merge(graphdef, params, counts)
 # update with multiple States
 nnx.update(model, params, counts)
 ```
+
+## Using Modules as Pytrees
+
+Even though `nnx.split` and `nnx.merge` can be used to interact with any JAX
+API, they are not always the most convenient way to do so as they introduce
+some syntactic overhead. `Module`s and other `Object`-derived types as PyTrees
+via the `unsafe_pytree` class argument for convenience. This allows you to
+pass Modules directly to JAX functions without having to split them first.
+
+```{code-cell} ipython3
+class Block(nnx.Module, unsafe_pytree=True): # <== 👀 unsafe_pytree
+  def __init__(self, din: int, dout: int, *, rngs: nnx.Rngs):
+    self.linear = Linear(din, dout, rngs=rngs)
+    self.dropout = nnx.Dropout(rate=0.1, rngs=rngs)
+
+  def __call__(self, x: jax.Array):
+    return nnx.gelu(self.dropout(self.linear(x)))
+  
+model = Block(3, 5, rngs=nnx.Rngs(0))
+
+@jax.jit  # regular jax.jit!
+def forward(model: Block, x: jax.Array):
+  y = model(x)
+  return y, model  # manually propagate state updates
+
+y, model = forward(model, jnp.ones((1, 3)))
+```
+
+**WARNING**: The reason the features is called `unsafe` is because NNX's 
+reference semantics are broken by JAX's referential transparency, this 
+is specially problematic when there is shared state between NNX graph nodes 
+as reference identity is lost. Use `unsafe_pytree` only when there' i's only 
+a single top-level Module or there is no shared state between top-level Modules.
+
+```{code-cell} ipython3
+class Foo(nnx.Module, unsafe_pytree=True):
+  def __init__(self, shared):
+    self.shared = shared
+
+shared = nnx.Linear(3, 5, rngs=nnx.Rngs(0))
+ma, mb = Foo(shared), Foo(shared)
+
+print(f'Before: {ma.shared is mb.shared = }')
+
+# flatten + unflatten
+ma, mb = jax.tree.map(lambda x: x, (ma, mb))
+
+print(f'After:  {ma.shared is mb.shared = }')
+```
```

Comment:
@ASEM000 here is the explanation, indeed its because of sharing. The `unsafe` naming hopefully gives users a visual queue that they shouldn't use this feature lightly.

Reply:
It is more clear now, thanks.